src/OpenSpaceToolkit/Astrodynamics/Trajectory/State/CoordinatesSubsets/AngularVelocity.cpp

```diff
@@ -0,0 +1,63 @@
+/// Apache License 2.0
+
+#include <OpenSpaceToolkit/Core/Error.hpp>
+
+#include <OpenSpaceToolkit/Astrodynamics/Trajectory/State/CoordinatesSubsets/AngularVelocity.hpp>
+
+namespace ostk
+{
+namespace astro
+{
+namespace trajectory
+{
+namespace state
+{
+namespace coordinatessubsets
+{
+
+using ostk::math::object::Vector3d;
+
+using ostk::physics::coord::Position;
+using ostk::physics::coord::Velocity;
+
+AngularVelocity::AngularVelocity(const Shared<const AttitudeQuaternion>& aAttitudeQuaternionSPtr, const String& aName)
+    : CoordinatesSubset(aName, 3),
+      attitudeQuaternionSPtr_(aAttitudeQuaternionSPtr)
+{
+}
+
+AngularVelocity::~AngularVelocity() {}
+
+VectorXd AngularVelocity::inFrame(
+    const Instant& anInstant,
+    const VectorXd& aFullCoordinatesVector,
+    const Shared<const Frame>& fromFrame,
+    const Shared<const Frame>& toFrame,
+    const Shared<const CoordinatesBroker>& aCoordinatesBrokerSPtr
+) const
+{
+    const VectorXd coordinates = aCoordinatesBrokerSPtr->extractCoordinate(aFullCoordinatesVector, *this);
+
+    const VectorXd attitudeCoordinatesInFrame = this->attitudeQuaternionSPtr_->inFrame(
+        anInstant, aFullCoordinatesVector, fromFrame, toFrame, aCoordinatesBrokerSPtr
+    );
+    const Quaternion quaternionInFrame = AttitudeQuaternion::coordinatesToQuaternion(attitudeCoordinatesInFrame);
+
+    const Transform transform = fromFrame->getTransformTo(toFrame, anInstant);
```

Comment:
Need to check something here, hold on

Reply:
So I understand you are using the attitude quaternion in here to be able to perform the Frame conversion, similar to what is being done in [here](https://github.com/open-space-collective/open-space-toolkit-astrodynamics/blob/main/src/OpenSpaceToolkit/Astrodynamics/Flight/Profile/State.cpp#L182). I think there are two steps here: 1. the frame in which the angular velocity is expressed and 2. the frame of reference relative to which the angular velocity is represented that come into play here. Do you agree?

Reply:
So basically the term using the attitude quaternion is related to the fact that the angular velocity from `fromFrame` to `toFrame` obtained from the `transform` is not expressed in the same frame as the others (potentially). IMO this is dealing with two different things: the expression is correct when we make the assumption that we express the input angular velocity coordinates in body frame _I think_ Totally fine by me but same as our discussion earlier: we should make that clear and IMO ultimately one is linked with to the Frame in which a vector (like all vectors) would be expressed (same as Velocity or Position for instance here) and the other is really a change of reference frame (which is more aligned with the `inFrame` method purpose IMO), so we might want to reflect this. For now though, assuming that it is expressed in body frame seems legit to me

Reply:
(Basically, the addition of two angular velocity vectors expressed in two different frames should be caught when adding them, in the same way that you can't add the coordinates of two Velocity vectors if not expressed in the same Frame). That's a detail, the expression here is correct for our current usage but pointing this out if we want to make this subset more generic in the future

Reply:
(and historically that's linked to `const Vector3d& anAngularVelocity` and not being a construct of its own, involving a Frame I'd say)

Reply:
Discussed with @phc1990:
- Fundamentally, attitude and angular velocity go one step further because they are considering 3D body dynamics
- Ideally, just like we have `physics` concepts `Position`, `Velocity` we should in the future rely on `Attitude` (which would have a type like `Quaternion`, `RotationMatrix` etc but also a frame of reference), some of the frame conversion could then be handled with that intermediate object `Quaternion` is simply a `mathematics` concept with no knowledge of `Frame` and cannot fulfill this completely, instead of having some of these conversions done in the coordinateSubset because angular velocity is a simple Vector and attitude a simple Quaternion
- For now, we can get away with this ✅